catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py

```diff
@@ -0,0 +1,175 @@
+    elif match_entry.field == "protocolName":
+        return ProtocolNameMatch.from_protocol_name_list(cast_(match_entry.value.split(" ")))
+    elif match_entry.field == "protocol":
+        return Protocol.from_protocol_id_list(match_entry.value.split(" "))
+
+    # TODO:
+    # SourceIdentityList, SourceSecurityGroup, DestinationSecurityGroup
+    # SourceIdentityUser,SourceIdentityUserGroup, App, AppFamily
+    convert_result.update_status(
+        "partial", f"Unknown conversion of firewall sequence match entry with type: {match_entry.field}"
+    )
+    return None
+
+
+def convert_sequence_match(match, convert_result: ConvertResult[NgfirewallParcel]):
+    entries = [convert_sequence_match_entry(entry, convert_result) for entry in match.entries]
+    return Match(entries=[entry for entry in entries if entry is not None])
+
+
+def convert_sequence_actions(
+    actions: List[Union[LogActionV1, AdvancedInspectionProfileAction, ConnectionEventsAction]],
+    convert_result: ConvertResult[NgfirewallParcel],
+) -> List[Union[LogAction, AipAction]]:
+    converted_actions = [convert_sequence_action_entry(action, convert_result) for action in actions]
+    filtered_actions = [action for action in converted_actions if action is not None]
+    return filtered_actions
+
+
+def convert_sequence_action_entry(
+    action: ActionEntry, convert_result: ConvertResult[NgfirewallParcel]
+) -> Optional[Union[LogAction, AipAction]]:
+    if action.type == "log":
+        return LogAction.from_sequence_action("log")
+    elif action.type == "advancedInspectionProfile":
+        return AipAction.from_uuid(action.parameter.ref)
+    elif action.type == "connectionEvents":
+        return LogAction.from_sequence_action("connectionEvents")
+
+    convert_result.update_status("partial", f"Unknown conversion of firewall sequence action type: {type(action)}")
+    return None
+
+
+def convert_zone_based_fw(
+    in_: ZoneBasedFWPolicy, uuid: UUID, context: PolicyConvertContext
+) -> ConvertResult[NgfirewallParcel]:
+    convert_result = ConvertResult[NgfirewallParcel]()
+
+    if in_.definition.entries:
+        context.zone_based_firewall_residues[uuid] = in_.definition.entries
+
+    try:
+        sequences = [
+            NgFirewallSequence.create(
+                **sequence.model_dump(exclude={"sequence_ip_type", "ruleset", "sequence_type", "match", "actions"}),
+                match=convert_sequence_match(sequence.match, convert_result),
+                actions=convert_sequence_actions(sequence.actions, convert_result),
+            )
+            for sequence in in_.definition.sequences
+        ]
+
+        parcel = NgfirewallParcel.create(
+            parcel_name=in_.name,
+            parcel_description=in_.description,
+            default_action_type=in_.definition.default_action.type,
+            sequences=sequences,
+        )
+        convert_result.output = parcel
+
+    except Exception as e:
```

Comment:
The broad `except Exception` clause could be more specific to catch expected exceptions.

Reply:
looks like `ValueError` was added to come construction methods